test/unittests/UtilityTest.php

```diff
@@ -858,4 +858,37 @@ public function testValueIsPositiveIntegerReturnsTrue($int): void
     {
         $this->assertTrue(\Utility::valueIsPositiveInteger($int));
     }
+
+    /**
+     * Tests the pathJoin function. Test cases adapted from blog post on
+     * Python's os.path.join as this function is meant to give the same
+     * beahviour.
```

Comment:
The behaviour of the edge cases you're using in this test don't make sense to me and are far less intuitive than Go's `filepath.Join`. Just because Python does something doesn't mean it's the right way to do it.

Reply:
OK, you originally asked for this in another PR which I quoted to make the issue. The link to the original PR seems to be broken so I'm not sure of the exact context.

Should I change this to behave like Go then?